### ISSUE TITLE
Dockerfile in ".devcontainer" is redundant and slow to build

### ISSUE THREAD

**Issue by KEINOS** [KEINOS/TPL-PHP-HelloWorld#60]
Currently, the Dockerfile in `.devcontainer` uses `php:7-cli` and sets up what is needed.

https://github.com/KEINOS/TPL-PHP-HelloWorld/blob/1f06c7f8e42137aa02c29dbee901a03a2e3b241b/.devcontainer/Dockerfile#L6

But, using the official PHP image of VSCode's Remote-Containers contains all the settings and saves a lot of time to build and maintain.

Here's the Dockerfile which VSCode's Remote-Containers generated by default.

```dockerfile
ARG VARIANT="7"

FROM mcr.microsoft.com/vscode/devcontainers/php:0-${VARIANT}
ARG INSTALL_NODE="true"
ARG NODE_VERSION="lts/*"
RUN if [ "${INSTALL_NODE}" = "true" ]; then su vscode -c "source /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
```

The above container contains `git` `wget` `zip` `unzip` `locales` `composer 2.0.6` and etc. Even `xdebug` is enabled and has the user `vscode` by default.

**Comment by KEINOS** [comment]
We still need the "[setup_composer.sh](https://github.com/KEINOS/TPL-PHP-HelloWorld/blob/1f06c7f8e42137aa02c29dbee901a03a2e3b241b/.devcontainer/setup_composer.sh)" for Travis CI though.

**Comment by KEINOS** [comment]
Yet, another sample.

```dockerfile
# [Choice] PHP version: 7, 7.4, 7.3
ARG VARIANT="7"

# See here for image contents:
# https://github.com/microsoft/vscode-dev-containers/tree/v0.148.1/containers/php/.devcontainer/base.Dockerfile
FROM mcr.microsoft.com/vscode/devcontainers/php:0-${VARIANT}

# Install Node.js(Needs for some extensions)
ARG INSTALL_NODE="true"
ARG NODE_VERSION="lts/*"
RUN if [ "${INSTALL_NODE}" = "true" ]; then su vscode -c "source /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi

# Install additional OS packages.
# RUN apt-get update \
#     && export DEBIAN_FRONTEND=noninteractive \
#     && apt-get -y install --no-install-recommends \
#       <your-package-list-here>

# Update PHP Composer
RUN sudo composer self-update

# Install BOX3 and set path to composer bin as vscode user
USER vscode
RUN whoami \
  && composer global init --stability=dev --quiet \
  && cd ~/.composer \
  && composer require \
    humbug/box:@stable \
    phan/phan:@stable \
    php-coveralls/php-coveralls:@stable \
    phpbench/phpbench:dev-master \
    phpmd/phpmd:@stable \
    phpstan/extension-installer:@stable \
    phpstan/phpstan:@stable \
    phpunit/phpunit:@stable \
    psalm/phar:@stable \
    squizlabs/php_codesniffer:@stable \
  && echo 'export PATH="${HOME}/.composer/vendor/bin:${PATH}"' >> ~/.bashrc

ENV COMPOSER='composer.dev.json'

# [Optional] Uncomment this line to install global node packages.
# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
```

**Comment by KEINOS** [comment]
It uses composer v2 which has some issues with packages such as box3. Better fix the composer version to 1.10.17

```dockerfile
# Update PHP Composer
RUN sudo composer self-update 1.10.17
```
